enumerable.rb

```diff
@@ -0,0 +1,114 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each(_array = nil)
+    return to_enum unless block_given?
+
+    Array(self).length.times do |i|
+      yield(Array(self)[i])
+    end
+    self
+  end
+
+  def my_each_with_index(_array = nil)
+    return to_enum unless block_given?
+
+    Array(self).length.times do |index|
+      yield(to_a[index], index)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
+    end
+    result
+  end
+
+  def my_all?(obj = nil)
+    return true if obj.nil? && !block_given?
+
+    result = true
+    if obj.nil?
```

Comment:
- [ ] my_all, when no block or argument is given, should return false if any of the array elements passed is false yours returns true.